v2/gcs-to-splunk/src/main/java/com/google/cloud/teleport/v2/templates/GCSToSplunk.java

```diff
@@ -0,0 +1,253 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import com.google.cloud.teleport.v2.coders.FailsafeElementCoder;
+import com.google.cloud.teleport.v2.coders.SplunkEventCoder;
+import com.google.cloud.teleport.v2.transforms.CsvConverters;
+import com.google.cloud.teleport.v2.transforms.ErrorConverters.LogErrors;
+import com.google.cloud.teleport.v2.transforms.SplunkConverters;
+import com.google.cloud.teleport.v2.transforms.SplunkConverters.SplunkOptions;
+import com.google.cloud.teleport.v2.values.FailsafeElement;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.splunk.SplunkEvent;
+import org.apache.beam.sdk.io.splunk.SplunkIO;
+import org.apache.beam.sdk.io.splunk.SplunkWriteError;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.Flatten;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link com.google.cloud.teleport.v2.templates.PubSubToSplunk} pipeline is a streaming
+ * pipeline which ingests data from Cloud Pub/Sub, executes a UDF, converts the output to {@link
+ * org.apache.beam.sdk.io.splunk.SplunkEvent}s and writes those records into Splunk's HEC endpoint.
+ * Any errors which occur in the execution of the UDF, conversion to {@link
+ * org.apache.beam.sdk.io.splunk.SplunkEvent} or writing to HEC will be streamed into a Pub/Sub
+ * topic.
+ *
+ * <p>NOTE: This is a work in progress, do not attempt to run this pipeline
+ */
+public class GCSToSplunk {
+
+  /** String/String Coder for FailsafeElement. */
+  private static final FailsafeElementCoder<String, String> FAILSAFE_ELEMENT_CODER =
+      FailsafeElementCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of());
+
+  /** The tag for the headers of the CSV if required. */
+  static final TupleTag<String> CSV_HEADERS = new TupleTag<String>() {};
```

Comment:
I don't think these are intended to be used anywhere other than the template. If so, please make them private. If they _must_ be package-private for testing purposes, please add the `@VisibleForTesting` annotation. (This is generally discouraged but not always avoidable.)

Reply:
I'm not sure how to validate a `PCollectionTuple`'s contents without accessing these `TupleTag`, so I used the `@ VisibleForTesting`annotation instead of marking them private.